web/src/engine/websites/MultPorn.ts

```diff
@@ -0,0 +1,107 @@
+import { Tags } from '../Tags';
+import icon from './MultPorn.webp';
+import { Chapter, DecoratableMangaScraper, type Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise( resolve => {
+        const pages = [];
+        for (i = 1; i <= jcgal.getImageCount(); i++) {
+            pages.push(jcgal.getImageInfo(i).largeImageURL);
+        }
+        resolve(pages);
+    });
+`;
+
+const categories = ['comics', 'pictures', 'hentai', 'hentai_manga', 'rule_63', 'humor', 'gay_porn_comics'];
+
+type DrupalSettings = {
+    views?: {
+        ajaxViews?: {
+            [id: string]: {
+                pager_element: number,
+                view_args: string,
+                view_base_path: string,
+                view_display_id: string,
+                view_dom_id: string,
+                view_name: string,
+                view_path: string
+            }
+        }
+    }
+}
+
+type DrupalResult = {
+    command: string,
+    data :string
+}
+
+@Common.MangaCSS(new RegExp(`^{origin}/(${categories.join('|')})/[^/]+`), 'div.breadcrumb span.last')
+@Common.PagesSinglePageJS(pageScript, 1500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = `${this.URI.origin}/views/ajax`;
+
+    public constructor() {
+        super('multporn', 'MultPorn', 'https://multporn.net', Tags.Media.Comic, Tags.Media.Manga, Tags.Language.Multilingual, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let category of categories) {
+            const mangas = await Common.FetchMangasMultiPageCSS.call(this, provider, category + '?page={page}', 'div.view-content table tr td strong a', 0);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const url = new URL(manga.Identifier, this.URI);
+        const settings = await FetchWindowScript<DrupalSettings>(new Request(url), 'Drupal.settings');
+        const chapterList = [];
```

Comment:
This mutable variable is only used inside the `if` block, may be moved into its corresponding scope